ipp-v3-java-devkit/src/main/java/com/intuit/ipp/query/GenerateQuery.java

```diff
@@ -64,53 +65,61 @@ public final class GenerateQuery {
 	private GenerateQuery() {
 	}
 
-	/**
-	 * Method to create the query entity for the given class
-	 * 
-	 * @param cl the class
-	 * @return the proxified object
-	 */
+
 	@SuppressWarnings("unchecked")
 	public static <T> T createQueryEntity(Class<T> cl) {
-		Enhancer enhancer = new Enhancer();
+		Class<?> proxied = null;
 		if (cl.isInterface()) {
 			LOG.debug("The given class is interface");
-			//enhancer.setInterfaces(new Class[] { cl });
-			//enhancer.setCallback(new MyMethodInterceptor());
 		} else {
-			enhancer.setSuperclass(cl);
+			proxied = new ByteBuddy()
+					.subclass(cl)
+					.method(ElementMatchers.not(ElementMatchers.isClone().or(ElementMatchers.isFinalizer()).or(ElementMatchers.isEquals()).or(ElementMatchers.isHashCode()).or(ElementMatchers.isToString())))
+					.intercept(MethodDelegation.to(new MyMethodInterceptor()))
+					.make()
+//					.load(cl.getClassLoader(), ClassLoadingStrategy.Default.WRAPPER)
+					.load(ClassLoader.getSystemClassLoader(), ClassLoadingStrategy.Default.WRAPPER)
```

Comment:
We do still need to understand why `ClassLoader.getSystemClassLoader` works and `cl.getClassLoader` doesn't work. It could be an issue for some clients.